### PR TITLE
Add driver ebuilds for MFC-7820n, along with other ebuilds (scanner only) for models in brscan2 family

### PR DESCRIPTION
**Pull request by rpimonitrbtch**

### CODE REVIEW

**Comment by stefan-langenmaier** [comment]
Hello! First of all thanks for this massive pull request!

Is it okay if I review the commits first, as I don't have a way to test this change? :)

Tell me if you don't have the time to do the changes or in case I'm to nitpicky. :)

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Copyright 1999-2016 Gentoo Foundation
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: Until 2017

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: No longer needed

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+EAPI=5
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: Can it be raised to EAPI=6

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	fi
+
+	linux-info_pkg_setup
+}
+
+src_unpack() {
+	rpm_unpack ${A}
+}
+
+src_prepare() {
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: The whole function can then be removed.

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	chmod 755 "${D}/usr/local/Brother/lpd" || die
+	chmod 755 "${D}/usr/local/Brother/inf" || die
+	chmod 755 "${D}/usr/local/Brother" || die
+
+	chmod 755 "${D}/usr/local/Brother/cupswrapper" || die
+
+	mkdir -p "${D}/var/spool/lpd/MFC7820N" || die
+}
+
+pkg_postinst() {
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: This should be in install with relative symlinks.

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	"/usr/local/Brother/cupswrapper/cupswrapperMFC7820N-${PV}" || die
+	( ln -s "/usr/lib64/cups/filter/brlpdwrapperMFC7820N" "/usr/libexec/cups/filter/brlpdwrapperMFC7820N" ) || die
+
+	einfo "Brother MFC-7820N printer installed"
+}
+
+pkg_prerm() {
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: This won't be needed with relative symlinks in the install phase.

File: net-print/brother-mfc7820n-bin/metadata.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<!--
+# $Id$
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/metadata.xml]: No longer needed.

File: generator/brother-generator.sh [hunk 8]
```diff
@@ -12,6 +12,45 @@
...
 DEBUGGING="true"
 
+# Configuration for brscan2
+BRSCAN2_COMMAND="brsaneconfig2 -q"
+BRSCAN2_OUTDIR="brscan2"
+BRSCAN2_NAME="brother-%id%-bin"
+BRSCAN2_VERSION="1.0"
+BRSCAN2_CATEGORY="media-gfx"
+read -d '' BRSCAN2_EBUILD << EOD
+# Copyright 1999-2016 Gentoo Foundation
```
Comment (stefan-langenmaier) [inline generator/brother-generator.sh]: Until 2017

File: generator/brother-generator.sh [hunk 9]
```diff
@@ -12,6 +12,45 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# \$Id\$
```
Comment (stefan-langenmaier) [inline generator/brother-generator.sh]: No longer needed

File: generator/brother-generator.sh [hunk 10]
```diff
@@ -12,6 +12,45 @@
...
+	einfo "- using hostname:   /usr/local/Brother/sane/brsaneconfig2 -a name=%model% model=%model% nodename=<hostname>"
+	einfo "- using IP address: /usr/local/Brother/sane/brsaneconfig2 -a name=%model% model=%model% ip=<IP address>"
+}
+EOD
+
+read -d '' BRSCAN2_METADATA << EOD
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<!--
+# \$Id\$
```
Comment (stefan-langenmaier) [inline generator/brother-generator.sh]: No longer needed

File: media-gfx/brother-scan2-bin/brother-scan2-bin-0.2.5-r1.ebuild [hunk 11]
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2016 Gentoo Foundation
```
Comment (stefan-langenmaier) [inline media-gfx/brother-scan2-bin/brother-scan2-bin-0.2.5-r1.ebuild]: Until 2017

File: media-gfx/brother-scan2-bin/brother-scan2-bin-0.2.5-r1.ebuild [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
```
Comment (stefan-langenmaier) [inline media-gfx/brother-scan2-bin/brother-scan2-bin-0.2.5-r1.ebuild]: No longer needed

**Comment by rpimonitrbtch** [comment]
I'm sure I'll be able to find some time. Doesn't look like the changes you're asking for are too crazy. I can be pretty nitpicky too.

File: net-print/brother-mfc7820n-bin/metadata.xml (hunk 7 above)
Comment (rpimonitrbtch) [reply net-print/brother-mfc7820n-bin/metadata.xml]: Is this (and the other such comments) referring to the "$Id$" lines?

Comment (stefan-langenmaier) [reply net-print/brother-mfc7820n-bin/metadata.xml]: Yes, the _$Id$_ is an artifact from the time before the move to git. So the whole comment can be removed.

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild (hunk 4 above)
Comment (stefan-langenmaier) [reply net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: The prepare function is only needed if there is a patch. So the whole src_prepare can be removed.

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+inherit eutils rpm linux-info
+
+DESCRIPTION="Brother printer driver for MFC-7820N"
+
+HOMEPAGE="http://support.brother.com"
+
+MY_PR=${PR/r/}
+
+SRC_URI="http://download.brother.com/welcome/dlf006265/brmfc7820nlpr-${PV}-${MY_PR}.i386.rpm
+        http://download.brother.com/welcome/dlf006267/cupswrapperMFC7820N-${PV}-${MY_PR}.i386.rpm"
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: Can you replace the spaces to a tab. Repoman prefers it that way.

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+
+
```
Comment (stefan-langenmaier) [inline net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: Also a comment from repoman. Remove the empty line. One empty line is enough. :)

File: net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild (hunk 4 above)
Comment (rpimonitrbtch) [reply net-print/brother-mfc7820n-bin/brother-mfc7820n-bin-2.0.1-r1.ebuild]: Yeah, but there is a patch now. Moving the postinst stuff into install, the cupswrapper script needs a patch to install to the sandbox.

**Comment by stefan-langenmaier** [comment]
Thanks again for all your work!

**Pull request merged**